In [1]:
import pandas as pd

In [3]:
pd.set_option('display.max_columns', 100)

# 01 — Data Inspection

**Dataset:** ICPSR Study 02538 — Uniform Crime Reporting (UCR), 1960–1962
**Source:** National Archive of Criminal Justice Data (NACJD)
**Format:** Stata `.dta`, one file per year
**Purpose of this notebook:** Read-only audit of the raw data. No cleaning, no transformation.

In [9]:
# importing DataSets
"""
1960 - df60
1961 - df61
1962 - df62
"""
df60 = pd.read_stata('../Data/Raw/02538-0002-Data1960.dta')
df61 = pd.read_stata('../Data/Raw/02538-0004-Data1961.dta')
df62 = pd.read_stata('../Data/Raw/02538-0006-Data1962.dta')

In [13]:
# Shapes of the DataFrame
print(df60.shape)
print(df61.shape)
print(df62.shape)

(20560, 72)
(23232, 72)
(24696, 72)


## Load Confirmation

- 1960: shape (20560, 72)
- 1961: shape (23232, 72)
- 1962: shape (24696, 72)
- All three share the same 72 columns in the same order
- Row counts grow across years — reporting coverage expanded

In [14]:
print("1960"); print(df60.dtypes.value_counts())
print()
print("1961"); print(df61.dtypes.value_counts())
print()
print("1962"); print(df62.dtypes.value_counts())

1960
int32       60
str          3
int16        2
float64      1
category     1
category     1
category     1
category     1
category     1
category     1
Name: count, dtype: int64

1961
int32       60
str          3
int16        2
float64      1
category     1
category     1
category     1
category     1
category     1
category     1
Name: count, dtype: int64

1962
int32       60
str          3
int16        2
float64      1
category     1
category     1
category     1
category     1
category     1
category     1
Name: count, dtype: int64
